### PR TITLE
soc: stm32wl: Adding logging support to stm32wl soc

### PR DESCRIPTION
**Pull request by AnachronisticPenguin**
As described in issue [#57655](https://github.com/zephyrproject-rtos/zephyr/issues/57655), the `STM32WL` SOC functions in `soc/arm/st_stm32/stm32wl` are missing a call to register the module with the logging module, which are needed by the power state specific code in `power.c` when logging is at the `DEBUG` level. Attempting to compile an application built for this SOC with `CONFIG_LOG_DEFAULT_LEVEL=4` will result in a compile-time error.

This commit adds a call to `LOG_MODULE_REGISTER` to the `soc.c` file which resolves the error.

### CODE REVIEW

**Review by erwango — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]

**Comment by erwango** [comment]
@AnachronisticPenguin please fix compliance issues reported [here](https://github.com/zephyrproject-rtos/zephyr/actions/runs/4946669198/jobs/8846627244?pr=57782#step:9:28)

**Comment by AnachronisticPenguin** [comment]
> @AnachronisticPenguin please fix compliance issues reported [here](https://github.com/zephyrproject-rtos/zephyr/actions/runs/4946669198/jobs/8846627244?pr=57782#step:9:28)

Made the changes. I think the two remaining workflow runs require maintainer approval to run.

File: soc/arm/st_stm32/stm32wl/soc.c
Comment (fabiobaltieri) [inline soc/arm/st_stm32/stm32wl/soc.c:3]: Is this a thing? Copyright headers don't normally get edits for every individual modifications, especially bug fixes like this one. I could understand major code reworks but I'd suggest dropping this in this case, imagine if everyone adds their line to every file, we'd have more copyrights than code :-)

Comment (AnachronisticPenguin) [reply soc/arm/st_stm32/stm32wl/soc.c:3]: I absolutely agree that it seems like overkill. But it seems to be required by section 4 of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0): "You must cause any modified files to carry prominent notices stating that You changed the files". This is repeated on the [WikiPedia page](https://en.wikipedia.org/wiki/Apache_License) for the license. It doesn't include language like "significant change" to provide an opt-out. 

See [here](https://softwareengineering.stackexchange.com/questions/220068/file-with-apache-2-0-and-my-modifications), [here](https://opensource.stackexchange.com/questions/8096/how-can-i-correctly-apply-the-apache-2-0-licence-to-contributed-code-from-an-exi), and [here](https://stackoverflow.com/questions/37781587/what-should-i-note-when-i-want-to-change-apache-license-2-0-code) for a few times where this has come up.

Comment (fabiobaltieri) [reply soc/arm/st_stm32/stm32wl/soc.c:3]: Interesting, did not realize it was spelled out like that in the licence, but I see what you mean:

https://github.com/zephyrproject-rtos/zephyr/blob/89bcb874db5271b4d6c7abb28e856275379ee875/LICENSE#L97-L98

@nashif @kartben do you we have a guidance for this? Wondering if it's been discussed before.

Comment (fabiobaltieri) [reply soc/arm/st_stm32/stm32wl/soc.c:3]: I mean, the Apache code itself does not have any of these: https://github.com/apache/httpd/tree/trunk/server

Comment (fabiobaltieri) [reply soc/arm/st_stm32/stm32wl/soc.c]: Thanks, it's a fair argument though and should probably be clarified, I'll keep asking around, thanks for pointing it out.

**Review by fabiobaltieri — APPROVED** [review]

File: soc/arm/st_stm32/stm32wl/soc.c
Comment (erwango) [reply soc/arm/st_stm32/stm32wl/soc.c]: I think that the mentioned line only concerns [redistribution](https://github.com/zephyrproject-rtos/zephyr/blob/89bcb874db5271b4d6c7abb28e856275379ee875/LICENSE#L89), which is not the current situation.
Current case is `5. Submission of Contributions`, , which isn't quite explicit about copyright notice. https://github.com/zephyrproject-rtos/zephyr/blob/89bcb874db5271b4d6c7abb28e856275379ee875/LICENSE#L130-L136

Comment (fabiobaltieri) [reply soc/arm/st_stm32/stm32wl/soc.c]: BTW can you reapprove @erwango?

**Review by erwango — APPROVED** [review]

**Pull request merged**
